src/it/java/EmulationITBase.java

```diff
@@ -0,0 +1,434 @@
+import org.metricshub.engine.telemetry.MonitorsVo;
+import org.metricshub.engine.telemetry.TelemetryManager;
+import org.metricshub.engine.telemetry.metric.AbstractMetric;
+import org.metricshub.extension.http.HttpExtension;
+import org.metricshub.extension.ipmi.IpmiExtension;
+import org.metricshub.extension.jdbc.JdbcExtension;
+import org.metricshub.extension.jmx.JmxExtension;
+import org.metricshub.extension.oscommand.OsCommandExtension;
+import org.metricshub.extension.ping.PingExtension;
+import org.metricshub.extension.snmp.SnmpExtension;
+import org.metricshub.extension.snmpv3.SnmpV3Extension;
+import org.metricshub.extension.wbem.WbemExtension;
+import org.metricshub.extension.winrm.WinRmExtension;
+import org.metricshub.extension.wmi.WmiExtension;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+public class EmulationITBase {
+
+	private static ExtensionManager extensionManager;
+	protected String connectorName;
+	protected TelemetryManager telemetryManager;
+	protected String configurationFilePath;
+
+	public EmulationITBase(final String connectorName, final String configurationFilePath) {
+		this.connectorName = connectorName;
+		this.configurationFilePath = configurationFilePath;
+	}
+
+	public EmulationITBase(final String connectorName) {
+		this(connectorName, "src/it/resources/" + connectorName + "/metricshub.yaml");
+	}
+
+	static void createExtensionManger() {
+		final List<IProtocolExtension> extensions = new ArrayList<>();
+		extensions.add(new HttpExtension());
+		extensions.add(new IpmiExtension());
+		extensions.add(new JdbcExtension());
+		extensions.add(new JmxExtension());
+		extensions.add(new OsCommandExtension());
+		extensions.add(new PingExtension());
+		extensions.add(new SnmpExtension());
+		extensions.add(new SnmpV3Extension());
+		extensions.add(new WbemExtension());
+		extensions.add(new WinRmExtension());
+		extensions.add(new WmiExtension());
+
+		extensionManager = ExtensionManager.builder()
+				.withProtocolExtensions(extensions)
+				.withConfigurationProviderExtensions(Collections.singletonList(new YamlConfigurationProvider()))
+				.build();
+
+	}
+
+	void emulateViaConfig() throws Exception {
+
+		// Create the extension manager
+		createExtensionManger();
+
+		// Set the connector emulation files, expected result and config directory
+		final String connectorFilesDirectory = Paths.get("src", "it", "resources", connectorName).toString();
+
+		// Initialize the application context
+		final ConfigReader configReader = new ConfigReader(

```

Comment:
Can we create an AgentContext please and then we remove the ConfigReader implementation?